contracts/tokens/SecurityToken.sol

```diff
@@ -527,11 +527,11 @@ contract SecurityToken is ERC20, ReentrancyGuard, SecurityTokenStorage, IERC1594
         // require(balanceOfByPartition(_partition, msg.sender) >= _value);
         // NB - Above condition will be automatically checked using the executeTransfer() function execution.
         // NB - passing `_additionalBalance` value is 0 because accessing the balance before transfer
-        uint256 balanceBeforeTransferLocked = _balanceOfByPartition(_partition, _to, 0);
+        uint256 lockedBalanceBeforeTransfer = _balanceOfByPartition(LOCKED, _to, 0);
```

Comment:
argument `_partition` is required to be `UNLOCKED` in `_isValidPartition` but it is not used in `_balanceOfByPartition` which instead hardcodes `LOCKED`. `toPartition` will always be `UNLOCKED` because of a bug in `_returnPartition`. finally an event gets emitted with the unused `_partition` (which needs to be `UNLOCKED` from the function call args.

I am struggling to understand the changes. @satyamakgec I would appreciate if you could elaborate on how this is working.

Reply:
So the Idea is that tokenHolder only allows transacting the `UNLOCKED` token. `_isValidPartition()` is used to validate whether the partition is UNLOCKED or not.  Then function checks the earlier `LOCKED` balance of the `_to` and after transfer, it will check the new `LOCKED` balance of `_to` so `_returnPartition` checks if the difference between both locked balance is equal to the value that is being transferred then it means all value is transferred to `LOCKED` partition. If not then it means some(or full) value transfferd to `UNLOCKED` partition of the receiver(_to).

Reply:
And I agree on the 542 line change I should add the else statement.